spec/lib/fastly_nsq/ssl_context_spec.rb

```diff
@@ -0,0 +1,74 @@
+require 'spec_helper'
+
+RSpec.describe SSLContext do
+  describe 'when SSL ENV variables are not set' do
+    describe '.to_h' do
+      it 'returns nil when initilized without parameters' do
+        context = SSLContext.new
+
+        expect(context.to_h).to be_nil
+      end
+
+      it 'returns a hash the same hash when all variables are defined' do
+        ssl_context_hash = {
+          key: 'key',
+          certificate: 'certificate',
+          ca_certificate: 'ca_certificate',
+        }
+        context = SSLContext.new(ssl_context_hash)
+
+        expect(context.to_h).to eq(ssl_context_hash)
+      end
+
+      it 'does not add keys with nil values' do
+        ssl_context_hash = {
+          key: 'key',
+          certificate: 'certificate',
+        }
+        context = SSLContext.new(ssl_context_hash)
+        ca_certificate = context.to_h[:ca_certificate]
+
+        expect(ca_certificate).to be_nil
+      end
+    end
+  end
+
+  describe 'when SSL ENV variables are set' do
+    before do
+      ENV['NSQ_SSL_KEY'] = '/some/key'
```

Comment:
So these changes will persist across tests when the `ENV` is mutated.

Instead, consider mocking the behavior out, as we do [here][1]:

```ruby
allow(ENV).to receive(:[]).with('FAKE_QUEUE').and_return(yes)
```

https://github.com/fastly/fastly_nsq/blob/master/spec/support/env_helpers.rb#L4